Use an InvWrapper's display name when possible

ShetiPhian opened the pull request:
This enables a different name to be provided for each InvWrapper, and enables a name to be displayed even if the TileEntity doesn't implement IWorldNameable.

bonii-xx commented on src/main/java/slimeknights/mantle/inventory/BaseContainer.java:
```diff
@@ -125,6 +126,12 @@ public boolean canInteractWith(@Nonnull EntityPlayer playerIn) {
 
   public String getInventoryDisplayName() {
+    if (itemHandler instanceof InvWrapper) {
+      ITextComponent textName = ((InvWrapper) itemHandler).getInv().getDisplayName();
```
Exctract variable for proper control flow

bonii-xx commented:
That.. was not exactly what I meant. Look at [the whole code](https://github.com/ShetiPhian/Mantle/blob/fd0cc9918809f204bc932942c960823826338261/src/main/java/slimeknights/mantle/inventory/BaseContainer.java#L130-L143). Lots of duplicate code there. DRY ;)

ShetiPhian commented:
Did I ever have a brain fail.
I had misread your response on my original request, and thought you wanted me to add the exact code I tested with. So when you asked for the variable to be extracted, I only looked at my added code and thought you wanted the line split for break points.
"Look at the whole code" was a face-palm moment for me, kinda yelled 'hey dummy integrate the code not just tack it on'

I cleaned that mess up and squashed the commits.

bonii-xx commented:
That's better. You don't need to squash manually anymore, github can do squash & merge nowadays
